src/poly/poly.jl

```diff
@@ -271,6 +271,17 @@ function Base.iterate(x::Nemo.Generic.MPolyMonomials{spoly{T}}, state) where T <
    end
 end
 
+###############################################################################
+#
+#   Hash functions
+#
+###############################################################################
+
+function Base.hash(p::spoly{T}) where T <: Nemo.RingElem
+   return 0 # FIXME: Implement a useful hash function instead
```

Comment:
You should overload
`Base.hash(::spoly{T}, h::UInt) where T<: Nemo.RingElem = h`

Reply:
See 754f514.